src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -146,6 +146,8 @@ interface ExpressionAstOptions {
   asDatatable?: boolean;
 }
 
+const omitByIsNil = <T>(object: Record<string, unknown>) => omitBy(object, isNil) as T;
```

Comment:
I think this will remove the need to cast using `as` below:

```suggestion
const omitByIsNil = <T>(object: Record<string, T>) => omitBy<T>(object, isNil);
```

The type will be inferred by whatever is passed into the function.